### ISSUE TITLE
MWoffliner should be provided as npm package

### ISSUE THREAD

**Issue by kelson42** [kiwix/mwoffliner#25]
A module with core function to dump a Mediawiki should be created, splitted from the runnable scripts and pubt npmjs.org

**Comment by lebouquetin** [comment]
Kelson42, after our talk on IRC, here is what we propose to do.

1. Refactor mwoffliner in two parts:
  - mwoffliner.js which is the "main" executable through command line
  - lib/mwoffliner.js which encapsulate the "brain" accessible through 2 functions:
    - setup(...) which allow to setup all arguments
   - execute() which run the process

2. publish mwoffliner as a npm package.

When importing mwoffliner through npm, a user will be able to use it through the following code : 

```
var mwo = require('mwoffliner')
mwo.setup(a, b, c, ...) // setup all config parameters
mwo.execute()
```

Optionnally the user will be able to run mwoffliner as a ready-to-use command-line (but it's not the goal of the npm publication)

**Comment by kelson42** [comment]
@lebouquetin @Skylsmoi Seem all in one to make it but the execution of the script seems to be still buggy.

After installing it on the system, I still can not execute it:
```
$ mwoffliner --mwUrl=https://en.wikipedia.org/ --adminEmail=foo@bar.net
Admin email undefined is not valid

```

Same for executing mwmatrixoffliner:
```
$ ./bin/mwmatrixoffliner.script.js --mwUrl=https://meta.wikimedia.org/ --adminEmail=kelson@kiwix.org --language=fr --project=wikiquote
Admin email undefined is not valid
Error by executing /home/kelson/mwoffliner/bin/mwoffliner.script.js

```

**Comment by Skylsmoi** [comment]
It has been fixed, see #51.

**Comment by kelson42** [comment]
@Skylsmoi Works for me :+1:
